staging/src/k8s.io/kubectl/pkg/cmd/cp/cp.go

```diff
@@ -109,7 +109,7 @@ func NewCmdCp(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.C
 			cmdutil.CheckErr(o.Run(args))
 		},
 	}
-	cmd.Flags().StringVarP(&o.Container, "container", "c", o.Container, "Container name. If omitted, the first container in the pod will be chosen")
+	cmd.Flags().StringVarP(&o.Container, "container", "c", o.Container, "Container name. If omitted, use this annotation to preselect container instead of the first container name in the pod as the default when kubectl.kubernetes.io/default-logs-container is specified.")
```

Comment:
My suggestion would be something like:
> Container name. If omitted, prefer to use `kubectl.kubernetes.io/default-container` annotation or the first container in the pod will be chosen

/cc @soltysh @rikatz
for some suggestions here.